### PR TITLE
feat: poc for apollo-router integation into bats setup

### PR DESCRIPTION
**Pull request by bodymindarts**

### CODE REVIEW

File: dev/apollo-federation/supergraph.graphql [hunk 1]
```diff
@@ -0,0 +1,1828 @@
+schema
```
Comment (nicolasburtey) [inline dev/apollo-federation/supergraph.graphql:1]: some README or pointer would be useful. I'm not too sure how to interpret this file right now without more context

File: docker-compose.yml [hunk 2]
```diff
@@ -53,6 +55,13 @@ services:
       - lnd-outside-3
       - otel-agent
     restart: on-failure:10
+  apollo-router:
+    image: ghcr.io/apollographql/router:v1.25.0
+    environment:
+      - APOLLO_ROUTER_SUPERGRAPH_PATH=/repo/dev/apollo-federation/supergraph.graphql
```
Comment (nicolasburtey) [inline docker-compose.yml:61]: I think we don't typically hard code stuff in here? instead we use env variable?

File: dev/apollo-federation/supergraph-config.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+federation_version: =2.3.2
+subgraphs:
+  main:
```
Comment (nicolasburtey) [inline dev/apollo-federation/supergraph-config.yaml:3]:
```suggestion
  public:
```

Comment (nicolasburtey) [reply dev/apollo-federation/supergraph-config.yaml:3]: we want to converge to `public` instead of `main` I think.

File: docker-compose.yml (hunk 2 above)
Comment (bodymindarts) [reply docker-compose.yml:61]: we do actually hard code paths to config.
See Otel agent for example:
```
  otel-agent:
    image: otel/opentelemetry-collector-contrib:0.79.0
    command: ["--config=/etc/otel-agent-config.yaml"]
    environment:
      - HONEYCOMB_DATASET=${HONEYCOMB_DATASET}
      - HONEYCOMB_API_KEY=${HONEYCOMB_API_KEY}
    volumes:
      - ./dev/otel-agent-config.yaml:/etc/otel-agent-config.yaml
```

File: dev/apollo-federation/supergraph-config.yaml (hunk 3 above)
Comment (bodymindarts) [reply dev/apollo-federation/supergraph-config.yaml]: done https://github.com/GaloyMoney/galoy/pull/2940/commits/aeed158af97cbea43267dd6bb6a5ab123a0caaea

File: dev/apollo-federation/supergraph.graphql (hunk 1 above)
Comment (bodymindarts) [reply dev/apollo-federation/supergraph.graphql:1]: it gets generated in the `yarn write-sdl` task. Will think about where to document it.

Comment (bodymindarts) [reply dev/apollo-federation/supergraph.graphql:1]: Added a readme under `dev/apollo-federation`

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**